### PR TITLE
Project discovery improvements

### PR DESCRIPTION
**Pull request by medgyes**
- discovery settings are moved into a separate dataclass 
 - discovery settings can be set on the event data table level
 - if it is not configured then discovery runs with the project level settings

### CODE REVIEW

File: mitzu/adapters/databricks_adapter.py [hunk 1]
```diff
@@ -43,6 +43,7 @@ def map_type(self, sa_type: Any) -> M.DataType:
     def _parse_map_type(
         self, sa_type: Any, name: str, event_data_table: M.EventDataTable
     ) -> M.Field:
+        assert event_data_table.discovery_settings is not None
```
Comment (IstvanM) [inline mitzu/adapters/databricks_adapter.py:46]: please avoid production code asserts ... rather throw an exception. 
https://towardsdatascience.com/2-caveats-to-avoid-when-using-python-assert-statement-8060e123c26a

File: mitzu/model.py [hunk 2]
```diff
@@ -354,6 +354,22 @@ def password(self):
...
 
+@dataclass(frozen=True)
+class DiscoverySettings:
+    max_enum_cardinality: int = 300
+    max_map_key_cardinality: int = 300
+
+    end_dt: Optional[datetime] = None
+    property_sample_rate: int = 100
+
+    lookback_days: int = 2
```
Comment (IstvanM) [inline mitzu/model.py:365]: the default can be more, at least 14 days.

File: mitzu/model.py [hunk 3]
```diff
@@ -354,6 +354,22 @@ def password(self):
...
+
+
+@dataclass(frozen=True)
+class WebappSettings:
+    lookback_window: TimeWindow = TimeWindow(30, TimeGroup.DAY)
```
Comment (IstvanM) [inline mitzu/model.py:370]: very cool!!!! So we can follow this pattern for every default

File: mitzu/adapters/databricks_adapter.py (hunk 1 above)
Comment (medgyes) [reply mitzu/adapters/databricks_adapter.py]: https://github.com/mitzu-io/mitzu/pull/66/commits/b10141a2efd1923e4faa4333158ab57e2baa8170

File: mitzu/model.py (hunk 2 above)
Comment (medgyes) [reply mitzu/model.py]: https://github.com/mitzu-io/mitzu/pull/66/commits/bdaee34730fa0b63326651ebad6ddf01ea38a0f6

**Review by IstvanM — APPROVED** [review]
LGTM, let's merge this after 2022-11-30.

**Review by IstvanM — APPROVED** [review]

**Pull request merged**